docs/src/main/asciidoc/async-message-passing.adoc

```diff
@@ -0,0 +1,290 @@
+It offers 3 types of delivery mechanism:
+
+- point-to-point - send the message, one consumer receives it. If several consumers listen to the address, a round robin is applied;
+- publish/subscribe - publish a message, all the consumers listening to the address are receiving the message;
+- request/reply - send the message and expect a response. The receiver can respond to the message in an asynchronous-fashion
+
+All these delivery mechanism are non-blocking, and are providing one of the fundamental brick to build reactive applications.
+
+== Installing
+
+This mechanism uses the Vert.x EventBus, so you need to enable the `vertx` extension to use this feature.
+If you are creating a new project, set the `extensions` parameter are follows:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=vertx-quickstart \
+    -Dextensions="vertx"
+----
+
+If you have an already created project, the `vertx` extension can be added to an existing {project-name} project with
+the `add-extension` command:
+
+[source]
+----
+mvn quarkus:add-extension -Dextensions="vertx"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-vertx</artifactId>
+</dependency>
+----
+
+== Consuming events
+
+To consume events, use the `io.quarkus.vertx.ConsumeEvent` annotation:
+
+[source, java]
+----
+package org.acme.vertx;
+
+import io.quarkus.vertx.ConsumeEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class GreetingService {
+
+    @ConsumeEvent                           // <1>
+    public String consume(String name) {    // <2>
+        return name.toUpperCase();
+    }
+}
+----
+<1> If not set, the address is the fully qualified name of the bean, for instance, in this snippet it's `org.acme.NameService`
+<2> The method parameter is the message body. If the method returns _something_ it's the message response.
+
+// TODO Revisit this once the blocking PR is integrated - https://github.com/quarkusio/quarkus/pull/1646
+IMPORTANT: the code consuming the event must be _non-blocking_, as it's called on the Vert.x event loop.
+
+=== Configuring the address
+
+The `@ConsumeEvent` annotation can be configured to set the address:
+
+[source, java]
+----
+@ConsumeEvent("greeting")               // <1>
+public String consume(String name) {
+    return name.toUpperCase();
+}
+----
+<1> Receive the messages sent on the `greeting` address
+
+=== Consuming without replying
+
+In the previous examples, the consumer returns a result, but it's not necessary:
+
+[source,java]
+----
+@ConsumeEvent("greeting")
+public void consume(String event) {
+    // Do something with the event
+}
+----
+
+=== Replying
+
+The _return_ of a method annotated with `@ConsumeEvent` is used as response to the incoming message.
```

Comment:
You first give examples with returning a value, then do "consuming without replying" then document replying. I think you should have started with not having a reply and then documenting how to reply later, but not back and forth.

Reply:
This choice was made to be in sync with the quickstart. Trying to see if we can change it without having to change too much in the quickstart.

Reply:
Just don't change the quickstart ;)